### PR TITLE
✨(models) add edx enrollment events pydantic models

### PR DESCRIPTION
**Pull request by quitterie-lcs**
## Purpose

edx enrollment events are common events in tracking logs and easy to represent. For the validation and selection commands of ralph, they are one of the easiest to modelise.

## Proposal

- [x] `pydantic` models
- [x] edge case tests
- [x] `selector` tests

### CODE REVIEW

File: src/ralph/models/edx/enrollment.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    """Represents the `context` field of the `edx.course.enrollment.upgrade_clicked` server event.
+
+    In addition to the common context member fields,
+    this event  also includes the `mode` context member field.
+
+    Attribute:
+        mode (str): Consists of either the `audit` or `honor` value.
+    """
+
+    mode: Union[Literal["audit"], Literal["honor"]]
```
Comment (jmaupetit) [inline src/ralph/models/edx/enrollment.py:113]: Can you regroup fields to the top of the file?

File: src/ralph/models/edx/enrollment.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    """Represents the `context` field of the `edx.course.enrollment.upgrade.succeeded` server event.
+
+    In addition to the common context member fields,
+    this event  also includes the `mode` context member field.
+
+    Attribute:
+        mode (str): Consists of the `verified` value.
+    """
+
+    mode: Literal["verified"]
```
Comment (jmaupetit) [inline src/ralph/models/edx/enrollment.py:148]: Ditto.

**Review by jmaupetit — APPROVED** [review]
I only have a minor suggestion. :+1:
